src/utils/codec.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * @file Codec
+ * @name codec.ts<utils>
+ * @description |
+ *
+ *    Encoding base to emoji and converting to any formats.
+ *
+ *    Every time you import this file will auto-calcuting all emojis
+ *    instead of a big constants file.
+ */
+export const ALL_EMOJIS: string[] = (function () {
+    const range: number[][] = [
+        [128506, 80], // 80 - 128512 - 128592
+        [129293, 46], // 46 - 129293 - 123339
+    ]
+
+    return range
+        .map((v: number[], _: number) => {
+            return [...Array(v[1])].map((_: number, i: number) => String.fromCodePoint(v[0] + i))
+        })
+        .reduce((o, c) => o.concat(c))
+})()
+
+/**
+ * Binary class
+ *
+ * @property {string[]} bin - binary
+ */
+export class Binary {
+    private bin: string[]
+
+    /**
+     * Encode string to binary Array
+     *
+     * @param {string} b64 - Base64 string
+     */
+    constructor(b64: string) {
+        this.bin = atob(b64)
+            .split('')
+            .map((char: string) => {
+                return char.charCodeAt(0).toString(2)
+            })
```

Comment:
```suggestion
        this.bin = Array.from(atob(b64))
            .map((char: string) => char.charCodeAt(0).toString(2))
```

```
> "👋💊".split('')
< ["�", "�", "�", "�"]
> Array.from("👋💊")
< ["👋", "💊"]
```
see https://mdn.io/arrayfrom

Reply:
Yep, and the `atob` is a temporary solution because it can not `exec` in `nodejs`, I'll update a new version `Codec` class instead of `Binary` class and has `binary`, `hex`, `base64`, `base1024?` later tonight for extra usages.

Reply:
btw, this version's `constructor` passing `base64` string as arg, emojis are not counted :-P, but the `Array.from` is really reasonable.